src/textual/document/_document_navigator.py

```diff
@@ -101,6 +101,10 @@ def is_start_of_wrapped_line(self, location: Location) -> bool:
     def is_end_of_document_line(self, location: Location) -> bool:
         """True if the location is at the end of a line in the document.
 
+        Note that the "end" of a line is equal to its length (one greater
+        than the final index), since there is a space at the end of the line
+        for the cursor to rest.
+
```

Comment:
Thanks for taking #4145 on boardhere! Sorry I realise this PR still a draft, but just a reminder that this docstring might also need correcting here:

```
        Returns:
            True if and only if the document is on the last line of the document.
```

Reply:
Good spot!